sidecar/src/one_way_shared_memory.rs

```diff
@@ -124,6 +123,11 @@ where
                 unsafe { reinterpret_u8_as_u64_slice(handle.as_slice()) }.into();
             let new_generation = source_data.meta.generation.load(Ordering::Acquire);
 
+            if new_generation & 1 == 1 {
+                // mid-write
+                return (false, "".as_bytes());
```

Comment:
This should return `(false, skip_last_byte(cur_data.as_slice()));`, not an empty string (i.e. the current valid copy of data.)